department-of-veterans-affairs/vets-website
# MHV-57259 New PHR refresh flow

## Summary

- This PR introduces a fairly complicated new flow to alert users of the "currentness" of the records they are seeing. For a visual explanation of the flow, see the [Figma page](https://www.figma.com/file/mGZRdLypKGaFsHo5xp2kaZ/Medical-Records?type=design&node-id=2776-106324&mode=design&t=2Tobqpy0T647JBJ9-0).
- I have only implemented this for Care Summaries & Notes to start. This is a very large change and I want to break it up a bit.
- Added a new `isTimedOut` state to the refresh reducer. Turns to `true` after the status check polls for 2 minutes without an up-to-date status being reported.
- Added a new `updatedList` state to the records reducer. New records go into here and are only moved into the displayed record list on user request, or if the user browses away.

## Related issue(s)

### [MHV-57259](https://jira.devops.va.gov/browse/MHV-57259) - Update the PHR refresh UI changes for existing data flow ###

> Update the PHR refresh UI changes for existing data flow
> 
> Existing user for all MR domains.  PHR refresh may be working in the background and user will have access to existing data.  Once the  PHR refresh completes:
> 
> 1- No new data is found and message displays 
> 
> 2- New data is found for current domain and message displays giving the user the ability to reload to get new updated.
> 
> See Figma for designs and details of these PHR flows:
> 
> https://www.figma.com/file/mGZRdLypKGaFsHo5xp2kaZ/Medical-Records?type=design&node-id=2776-106324&mode=design&t=2Tobqpy0T647JBJ9-0 
> 
> 
> 
> 
> Feature Flag Y/N ? N
> 
> DataDog/Analytics?  Y/N ?
> 
> Manual Testing Y/N ? Y
> 
> Automated Testing Y/N ?
> 
> Accessibility Testing Y/N ? Y
> 
> UCD Validation Y/N Y
> 
> Acceptance Criteria:
> AC1 User is able to access existing data during a PHR refresh
> AC2 Message is displayed when no new data is found following FIGMA designs
> AC3 Message and reload button is displayed when new data for current domain is found following FIGMA designs
> AC4 UCD has validated changes (Amelia)
> AC5 The new PHR status content appears in the order specified in "Architectural Notes"
> 
> Links to UCD:
> Figma Link: https://www.figma.com/file/mGZRdLypKGaFsHo5xp2kaZ/Medical-Records?type=design&node-id=2776-106324&mode=design&t=2Tobqpy0T647JBJ9-0
> Other Design Notes
> 
> Architectural Notes:
> 
> The content will always appear in this order, regardless of which is actually present (per [MHV-58869](https://jira.devops.va.gov/browse/MHV-58869)):
> 
> 1. `<h1>` title
> 1. Subtitle/text description
> 1. New PHR refresh status component
> 1. Print/download button
> 1. What to know dropdown
> 1. Pagination and Record list

## Testing done

- Added a bunch of new unit tests for new functions
- Fixed existing tests to correct for new code

## Screenshots

### Refresh Already Happened ###
<img width="300" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/87040148/b4ce7648-2458-47be-8edd-42c1974115e5">

### Refresh in Progress ###
<img width="300" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/87040148/ed4816b8-c4cd-42cf-842c-da2046c568ba">

### Refresh Completed, No New Records ###
<img width="300" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/87040148/e4400ce2-6da1-4831-bebd-95a4b9dc0300">

### Refresh Completed, New Records Found ###
<img width="300" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/87040148/ec454781-1b03-4145-b545-d5442cf07fc2">

### Refresh Failed (e.g. Timed Out) ###
<img width="300" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/87040148/5ca3b019-9cb4-46e8-8051-a33979bad63d">


## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication
